src/gossip/gossip_service.zig

```diff
@@ -477,7 +491,7 @@ pub const GossipService = struct {
             const protocol_messages = maybe_protocol_messages.?;
             defer {
                 for (protocol_messages) |*msg| {
-                    bincode.free(self.allocator, msg.message);
+                    msg.message.shallowFree(self.allocator);
```

Comment:
can you add a comment talking about why we shallow free here (the pr description was great) - maybe also a `TODO: consider using reference counting`

Reply:
https://github.com/Syndica/sig/pull/69/commits/8dccb42101630fa7f7c9c966a11a69710fda7ffc